src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePumpSettings.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Threading.Tasks;
+using Arcus.Security.Core;
+using GuardNet;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Arcus.Messaging.Pumps.ServiceBus
+{
+    public class AzureServiceBusMessagePumpSettings
+    {
+        private readonly Func<ISecretProvider, Task<string>> _getConnectionStringFromSecretFunc;
+        private readonly Func<IConfiguration, string> _getConnectionStringFromConfigurationFunc;
+        private readonly IServiceProvider _serviceProvider;
+
+        /// <summary>
+        ///     Constructor
+        /// </summary>
+        /// <param name="entityName">Name of the entity to process</param>
+        /// <param name="subscriptionName">Name of the subscription to process</param>
+        /// <param name="getConnectionStringFromConfigurationFunc">Function to look up the connection string from the configuration</param>
+        /// <param name="getConnectionStringFromSecretFunc">Function to look up the connection string from the secret store</param>
+        /// <param name="options">Options that influence the behavior of the message pump</param>
+        /// <param name="serviceProvider">Collection of services to use</param>
+        public AzureServiceBusMessagePumpSettings(string entityName, string subscriptionName,
+            Func<IConfiguration, string> getConnectionStringFromConfigurationFunc,
+            Func<ISecretProvider, Task<string>> getConnectionStringFromSecretFunc,
+            AzureServiceBusMessagePumpOptions options, IServiceProvider serviceProvider)
+        {
+            Guard.For<ArgumentException>(
+                () => getConnectionStringFromConfigurationFunc == null && getConnectionStringFromSecretFunc == null,
+                "Unable to determine connection string as it was not defined how to look it up");
+            Guard.NotNull(options, nameof(options));
+            Guard.NotNull(serviceProvider, nameof(serviceProvider));
+
+            _serviceProvider = serviceProvider;
+            _getConnectionStringFromConfigurationFunc = getConnectionStringFromConfigurationFunc;
+            _getConnectionStringFromSecretFunc = getConnectionStringFromSecretFunc;
+
+            EntityName = entityName;
+            SubscriptionName = subscriptionName;
+            Options = options;
+        }
+
+        /// <summary>
+        ///     Name of the entity to process
```

Comment:
Microsoft code guidelines talk about starting XML docs for properties with `Get/Set ...`. I see that you combine both with and without this on the properties of this type. Maybe take an approach and stick with it. :)

Reply:
Can you elaborate on this please? Not sure if I'm getting it?

Where do I use another approach than this one?

Reply:
This is how Microsoft's guidelines are followed; starting the property XML docs with `Get or sets` (or only 'get' or 'set' with a getter-only prop of course).

```cshrap
public class Entity
{
    /// Gets or sets the name of the entity.
    public string Name { get; set; }
```

Now, in the code file (and mostly everywhere) this is not exactly followed, which is fine. Only a single property in this file follows this guideline, so I suggested to choose between the two. 😄